hazelcast/src/hazelcast/client/spi.cpp

```diff
@@ -1104,7 +1114,8 @@ namespace hazelcast {
                 }
 
                 bool ClientInvocation::is_bind_to_single_connection() const {
-                    return connection_ != nullptr;
+                    auto conn = connection_.lock();
```

Comment:
This does not feel right. 
There will be a connection managed by probably connection manager. When the connection is gone, this will return `false` even tough, the invocation was bing to single connection when created. 
I think it is better to replace this logic with a boolean which is decided in the constructor.

Reply:
Fixed. Added a bool flag `bound_to_single_connection_` to constructor.

Reply:
Resolving.